### PR TITLE
Clippy updates, let's box it

### PR DESCRIPTION
**Pull request by josojo**
Fixing clippy failures. Needs to be merged, such that other PRs turn green again, as CI is failing currently

### Test Plan
non

### CODE REVIEW

File: crates/shared/src/price_estimation/competition.rs [hunk 1]
```diff
@@ -55,18 +55,17 @@ impl PriceEstimating for CompetitionPriceEstimator {
 }
 
 #[derive(Debug)]
-struct EstimateData<'a> {
-    estimator_name: &'a str,
+struct EstimateData {
```
Comment (nlordell) [inline crates/shared/src/price_estimation/competition.rs:58]: Note that this change makes it so the `winning_estimate` log above no longer prints the estimator that was used.

**Review by nlordell — COMMENTED** [review]
Just minor issue with logging the winning estimate.

**Review by nlordell — APPROVED** [review]
Weird that it suggests boxing here. `clippy` is always right though ;)

**Comment by josojo** [comment]
>Weird that it suggests boxing here. clippy is always right though ;)

it's because the size for each enum  object is different, and it is a little bit more efficient, if we box it, so that we don't allocated the max object size.

**Pull request merged**